### DISCUSSION TITLE
[Artifacts API] Got Method Not Allowed when Create a public download URL

### DISCUSSION THREAD

**Discussion by arbyazra123** [codemagic-ci-cd/codemagic-docs#1418] [Q&A] [answered]
Hi, 
I tried to create public download url for specific build to customize the expiration time. 

I already followed the docs : https://docs.codemagic.io/rest-api/artifacts/  and tested in Postman and Terminal with curl but it keeps returning Method Not Allowed :

```{"error":"METHOD_NOT_ALLOWED","message":"The method is not allowed for the requested URL."}```

I already changed the method to GET, PUT and other methods but the error keep showing.
Thank you.

**Comment by icarusdust** [comment, marked as answer]
Hi @arbyazra123, it is because you are trying to set a date to a private link. You will need to do it with a publicly available artefact link. It would look like this:

`https://api.codemagic.io/artifacts/securepath/myapp.ipa/public-url`

**Reply by arbyazra123** [reply]
Hi, thank you for the follow up.

I already do what you did, but the problem is still the same

<img width="676" alt="Screen Shot 2023-02-20 at 14 34 48" src="https://user-images.githubusercontent.com/40289741/220042022-ff88abc2-fc6b-4e28-b580-c2b7b141d097.png">

**Reply by icarusdust** [reply]
I hope you are changing the link from `https://api.codemagic.io/artifacts/securepath/myapp.ipa/public-url `to your actual link, right ?

**Reply by arbyazra123** [reply]
i just tried with myapp.ipa (https://api.codemagic.io/artifacts/securepath/myapp.ipa/public-url), but there's no changes

**Reply by arbyazra123** [reply]
do you mean i need to set the build to public then i can generate the public link with custom expiration?

**Reply by icarusdust** [reply]
Your attached artefact url needs to be publicly downloadable as the doc states: https://docs.codemagic.io/rest-api/artifacts/#step-2-create-a-public-download-url-using-the-url-obtained-in-step-1

**Reply by arbyazra123** [reply]
I think i got the point, one last question, what does the :securePath means? @icarusdust

**Comment by arbyazra123** [comment]
Hi, finally i can make that public url. That's my fault, apparently it can be obtained from copying download link in codemagic with additional /public-url and x-auth-token in the headers. 
Thank youu @icarusdust
